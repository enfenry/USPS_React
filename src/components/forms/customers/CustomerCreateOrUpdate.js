import React from 'react';
import { Button, Form, FormGroup } from 'reactstrap';
import { reduxForm, Field } from 'redux-form';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import validate from './CustomerValidate';
import renderField from '../renderField';

let CustomerCreateOrUpdate = props => {
  const { handleSubmit, handleHide, addresses, initialValues } = props;

  const displayById = (array, key, value, display) => {
    let filtered = array.filter(el => el[key] === value);
    return filtered.length ? filtered[0][display] : 'None';
  }

  return (
    <Form onSubmit={handleSubmit} className="form">
      <FormGroup className="field">
        <div className="control">
          <Field name="firstname" component={renderField} type="text" label="First Name" />
        </div>
      </FormGroup>

      <FormGroup className="field">
        <div className="control">
          <Field name="lastname" component={renderField} type="text" label="Last Name" />
        </div>
      </FormGroup>

      <FormGroup className="field">
        <div className="control">
          <Field name="emailaddress1" component={renderField} type="text" label="Email" />
        </div>
      </FormGroup>

      <div>
        <div>
          <span>Address Name: </span>
          <p>{displayById(addresses, "ss_customaddressid", initialValues._ss_contactcustomaddress_value, "ss_name")}</p>
        </div>
      </div>

      <div className="field">
        <div className="control">
          <Button className="button is-link" color="success" onClick={handleSubmit}>Submit</Button>{' '}
          <Button color="secondary" onClick={handleHide}>Cancel</Button>
        </div>
      </div>
    </Form>
  );
};

CustomerCreateOrUpdate.propTypes = {
  handleSubmit: PropTypes.func,
  handleHide: PropTypes.func,
  initialValues: PropTypes.object,
  addresses: PropTypes.arrayOf(PropTypes.object),
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node
  ])
}

CustomerCreateOrUpdate = reduxForm({
  form: 'CustomerCreateOrUpdate',
  validate,
  onSubmitSuccess: (result, dispatch, props) => {
    props.handleHide();
  }
})(CustomerCreateOrUpdate);


function mapStateToProps(state) {
  return {
    addresses: state.addressesReducer.addresses
  }
}

export default connect(
  mapStateToProps
)(CustomerCreateOrUpdate);
