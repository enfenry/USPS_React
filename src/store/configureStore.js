import { createStore, applyMiddleware, compose } from 'redux';
import reduxImmutableStateInvariant from 'redux-immutable-state-invariant';
import thunk from 'redux-thunk';
import rootReducer from '../reducers';

import { createLogger } from 'redux-logger'
import DevTools from '../containers/DevTools';

function configureStore(initialState) {

  const logger = createLogger()

  const middlewares = [
    reduxImmutableStateInvariant(),
    thunk,
    logger
  ];

  const store = createStore(
    rootReducer,
    initialState,
    compose(
      applyMiddleware(...middlewares),
      DevTools.instrument()
    )
  );

  return store;
}

export default configureStore;
