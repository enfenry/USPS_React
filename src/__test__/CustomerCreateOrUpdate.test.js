import React from 'react';
import CustomerCreateOrUpdate from '../components/forms/customers/CustomerCreateOrUpdate';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';

import * as basicStateUtils from "./basicState";

import configureStore from 'redux-mock-store';
const mockStore = configureStore([]);

test('CustomerCreateOrUpdate renders correctly', () => {

    const state = basicStateUtils.getBasicState();
    const store = mockStore(state);
    const initialValues = state.customersReducer.customers[0];

    const component = renderer.create(
        <Provider store={store}>
            <CustomerCreateOrUpdate initialValues={initialValues} />
        </Provider>
    );

    let tree = component.toJSON();

    expect(tree).toMatchSnapshot();
});