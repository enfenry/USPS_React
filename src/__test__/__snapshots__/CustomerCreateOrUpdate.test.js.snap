// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomerCreateOrUpdate renders correctly 1`] = `
<form
  className="form"
  onSubmit={[Function]}
>
  <div
    className="field form-group"
  >
    <div
      className="control"
    >
      <div
        style={
          Object {
            "display": "block",
          }
        }
      >
        <div
          className="control"
        >
          <label
            className="field"
          >
            First Name
          </label>
          <input
            className="input form-control"
            name="firstname"
            onBlur={[Function]}
            onChange={[Function]}
            onDragStart={[Function]}
            onDrop={[Function]}
            onFocus={[Function]}
            type="text"
            value=""
          />
        </div>
      </div>
    </div>
  </div>
  <div
    className="field form-group"
  >
    <div
      className="control"
    >
      <div
        style={
          Object {
            "display": "block",
          }
        }
      >
        <div
          className="control"
        >
          <label
            className="field"
          >
            Last Name
          </label>
          <input
            className="input form-control"
            name="lastname"
            onBlur={[Function]}
            onChange={[Function]}
            onDragStart={[Function]}
            onDrop={[Function]}
            onFocus={[Function]}
            type="text"
            value=""
          />
        </div>
      </div>
    </div>
  </div>
  <div
    className="field form-group"
  >
    <div
      className="control"
    >
      <div
        style={
          Object {
            "display": "block",
          }
        }
      >
        <div
          className="control"
        >
          <label
            className="field"
          >
            Email
          </label>
          <input
            className="input form-control"
            name="emailaddress1"
            onBlur={[Function]}
            onChange={[Function]}
            onDragStart={[Function]}
            onDrop={[Function]}
            onFocus={[Function]}
            type="text"
            value=""
          />
        </div>
      </div>
    </div>
  </div>
  <div>
    <span>
      Address Name: 
    </span>
    <p>
      None
    </p>
  </div>
  <div
    className="field"
  >
    <div
      className="control"
    >
      <button
        aria-label={null}
        className="button is-link btn btn-success"
        onClick={[Function]}
        type="button"
      >
        Submit
      </button>
       
      <button
        aria-label={null}
        className="btn btn-secondary"
        onClick={[Function]}
      >
        Cancel
      </button>
    </div>
  </div>
</form>
`;
